File: app/src/main/java/ru/geekbrains/notes/card/CardSource.java
package ru.geekbrains.notes.card;

public interface CardSource {
    int size();
    CardData getCardData(int position);

    void deleteCardData(int position);
    void updateCardData(int position, CardData newCardData);
    void addCardData(CardData newCardData);
    void clearCardData();
}

File: app/src/main/java/ru/geekbrains/notes/repository/CardSource.kt
package ru.geekbrains.notes.repository

import ru.geekbrains.notes.model.CardData

interface CardSource {
    fun size(): Int
    fun getCardData(position: Int): CardData
    fun getCards(cardsSourceResponse: CardsSourceResponse): CardSource?
    fun deleteCardData(position: Int)
    fun updateCardData(position: Int, newCardData: CardData)
    fun addCardData(newCardData: CardData)
    fun clearCardData()
}
